apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/CaseFiles/CaseFilesForm.tsx

```diff
@@ -178,24 +179,86 @@ const CaseFilesForm: React.FC<Props> = (props) => {
         <Box marginBottom={5}>
           <AnimateSharedLayout>
             <motion.div layout className={styles.policeCaseFilesContainer}>
-              {policeCaseFileList.length > 0 ? (
-                <motion.ul layout>
-                  <motion.li
-                    layout
-                    className={cn(styles.policeCaseFile, {
-                      [styles.selectAllPoliceCaseFiles]: true,
-                    })}
+              <motion.ul layout>
+                <motion.li
+                  layout
+                  className={cn(styles.policeCaseFile, {
+                    [styles.selectAllPoliceCaseFiles]: true,
+                  })}
+                >
+                  <Checkbox
+                    name="selectAllPoliceCaseFiles"
+                    label={formatMessage(
+                      m.sections.policeCaseFiles.selectAllLabel,
+                    )}
+                    checked={checkAllChecked}
+                    onChange={(evt) => toggleCheckbox(evt, true)}
+                    disabled={isUploading || policeCaseFileList.length === 0}
+                    strong
+                  />
+                </motion.li>
+                {policeCaseFiles?.isLoading ? (
+                  <Box
+                    textAlign="center"
+                    paddingY={2}
+                    paddingX={3}
+                    marginBottom={2}
                   >
-                    <Checkbox
-                      name="selectAllPoliceCaseFiles"
-                      label={formatMessage(
-                        m.sections.policeCaseFiles.selectAllLabel,
+                    <LoadingDots />
+                  </Box>
+                ) : policeCaseFiles?.hasError ? (
+                  policeCaseFiles?.errorMessage &&
+                  policeCaseFiles?.errorMessage.indexOf('404') > -1 ? (
+                    <Box
```

Comment:
This box is duplicated 3 times depending on the error, wondering if we could make it more reusable.

Also these errors give the exact same warning box:
policeCaseFiles?.files.length === 0  and 
policeCaseFiles?.errorMessage && policeCaseFiles?.errorMessage.indexOf('404') > -1